src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -237,16 +447,26 @@ public String getName()
         {
             return this.name;
         }
-
-        public ITradeList[][] getTrades()
+        
+        public VillagerProfession getProfession()
         {
-            return this.trades;
+            return this.profession;
         }
 
-        private VillagerCareer init(EntityVillager.ITradeList[][] trades)
+        /**
+         * Retrieves the trades for the give skillLevel of the villager in this career.
+         * 
+         * @param skillLevel
+         *            the skill level
+         * @return An array containing all trades.
+         */
+        public ITradeList[] getTrades(int skillLevel)
         {
-            this.trades = trades;
-            return this;
+            List<List<ITradeList>> skilledTradeList = TRADES.get(this);
+            boolean validLevel = skillLevel >= 0 && skillLevel < skilledTradeList.size();
+            List<ITradeList> trades = Lists.newArrayList(skilledTradeList.get(validLevel ? skillLevel : 0));//To prevent tampering with the original values
+            MinecraftForge.EVENT_BUS.post(new VillagerTradePopulationEvent(this, trades, skillLevel));//Allows modders to modify the trade list.
+            return trades.toArray(new ITradeList[]{});
```

Comment:
I'm not convinced that there needs to be a special event for this to make each instance editable.
ANd if there is this is NOT the place to put it as this doesn't have ANY context beyond the global career info.
You'd want to edit the per-entity list not the global.

If you WANT to edit the global then you'd do something like Career c = registry("farmer").level(1); c.remove(trade); c.add(trade)